pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -39,7 +45,49 @@ type openShiftClusterDynamicValidator struct {
 
 	oc              *api.OpenShiftCluster
 	subscriptionDoc *api.SubscriptionDocument
-	fpAuthorizer    refreshable.Authorizer
+	fpAuthorizer    autorest.Authorizer
+}
+
+func ensureAccessTokenClaims(ctx context.Context, tokenCredential *azidentity.ClientSecretCredential, scopes []string) error {
+	var err error
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+
+	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		options := policy.TokenRequestOptions{Scopes: scopes}
+		token, err := tokenCredential.GetToken(ctx, options)
+		if err != nil {
+			return false, err
+		}
+
+		var claims jwt.MapClaims
+		parser := &jwt.Parser{UseJSONNumber: true}
+		_, _, err = parser.ParseUnverified(token.Token, &claims)
+		if err != nil {
+			err = api.NewCloudError(
+				http.StatusBadRequest,
+				api.CloudErrorCodeInvalidServicePrincipalToken,
+				"properties.servicePrincipalProfile",
+				"The provided service principal generated an invalid token.")
+			return false, err
+		}
+
+		for _, claim := range []string{"altsecid", "oid", "puid"} {
```

Comment:
Is there any documentation that can be linked to here about the significance of these 3 strings?

Reply:
Added a new permanent commit for this.